src/main/fc/tasks.c

```diff
@@ -158,23 +158,58 @@ static void taskUpdateAccelerometer(timeUs_t currentTimeUs)
 }
 #endif
 
+static enum {
+    CHECK, PROCESS, MODES, UPDATE
+} rxState = CHECK;
+
+bool taskUpdateRxMainInProgress()
```

Comment:
Yeah, this seems to be cleaner to me - except that now the `rxProcessPending()` calls can be removed as `processPending` isn't read any more. :smiley:

Reply:
Done. This is cleaner. You were right.